test/libsolidity/util/BytesUtils.cpp

```diff
@@ -202,21 +202,28 @@ string BytesUtils::formatString(bytes const& _bytes, size_t _cutOff)
 	return os.str();
 }
 
-string BytesUtils::formatRawBytes(bytes const& _bytes)
+string BytesUtils::formatRawBytes(bytes const& _bytes, string _linePrefix, bool _withSignature)
 {
 	if (_bytes.empty())
-		return "[]";
+		return _linePrefix + "[]";
 
 	stringstream os;
 	auto it = _bytes.begin();
-	for (size_t i = 0; i < _bytes.size(); i += 32)
+
+	if (_withSignature)
 	{
-		bytes byteRange{it, it + 32};
+		os << _linePrefix << bytes{it, it + 4} << endl;
+		it += 4;
+	}
 
-		os << "  " << byteRange;
+	bytes tail{it, _bytes.end()};
+	it = tail.begin();
 
+	for (size_t i = 0; i < tail.size(); i += 32)
```

Comment:
Doesn't this need `i + 32 <= tail.size()` if you use `it + 32` as end below?

Reply:
I've refactored the formatting a little bit and it's now all based on lists of `Parameter`. Please the last commit :)
This way we can at least assert that `_bytes.size() == encodingSize(parameters)` in functions that are supposed to format bytes in some way and avoid dealing with such iterator offsets altogether.